src/code/irqmgr.c

```diff
@@ -1,122 +1,174 @@
+/**
+ * @file irqmgr.c
+ *
+ * This file implements a manager for forwarding three key system interrupt events to
+ * registered clients. These interrupts are:
+ *
+ *  - VI Retrace
+ *      This event is sent to the IRQ manager by the OS VI manager which only supports
+ *      the forwarding of VI events to a single message queue. The IRQ manager will
+ *      forward these events to every registered client. VI retrace events are received
+ *      when the Video Interface has reached the start of the vertical blanking interval,
+ *      happening at approximately 60Hz on NTSC and 50Hz on PAL. Many threads sit idle
+ *      until a VI Retrace event wakes them up, at which point they will perform their
+ *      task and then return to idle to await the next retrace.
+ *
+ *  - Pre-NMI
+ *      This event is sent to the IRQ manager by the OS Interrupt Handler when the reset
+ *      button on the N64 control deck is pressed. This event is forwarded to clients to
+ *      inform them that a reset will occur in 0.5s / 500ms and to begin any shutdown
+ *      procedures.
+ *
+ *  - NMI
+ *      This event is sent at 450ms into the Pre-NMI phase, informing clients that a
+ *      reset is imminent.
+ */
 #include "global.h"
 #include "vt.h"
 
-vu32 gIrqMgrResetStatus = 0;
+vu32 gIrqMgrResetStatus = IRQ_RESET_IDLE;
 volatile OSTime sIrqMgrResetTime = 0;
 volatile OSTime gIrqMgrRetraceTime = 0;
 u32 sIrqMgrRetraceCount = 0;
 
+// Internal messages
 #define RETRACE_MSG 666
 #define PRE_NMI_MSG 669
 #define PRENMI450_MSG 671
 #define PRENMI480_MSG 672
 #define PRENMI500_MSG 673
 
-#define STATUS_IDLE 0
-#define STATUS_PRENMI 1
-#define STATUS_NMI 2
-
-void IrqMgr_AddClient(IrqMgr* this, IrqMgrClient* c, OSMesgQueue* msgQ) {
+/**
+ * Registers a client and an associated message queue with the irq manager. When an
+ * interrupt event is received by the irq manager, these clients will be notified of
+ * the event.
+ *
+ * @param this the IrqMgr instance to register with.
+ * @param client client to register.
+ * @param msgQ message queue to send notifications of interrupts to, associated with the client.
+ */
+void IrqMgr_AddClient(IrqMgr* this, IrqMgrClient* client, OSMesgQueue* msgQ) {
     OSIntMask prevInt;
 
     LogUtils_CheckNullPointer("this", this, "../irqmgr.c", 96);
-    LogUtils_CheckNullPointer("c", c, "../irqmgr.c", 97);
+    LogUtils_CheckNullPointer("c", client, "../irqmgr.c", 97);
     LogUtils_CheckNullPointer("msgQ", msgQ, "../irqmgr.c", 98);
 
     prevInt = osSetIntMask(OS_IM_NONE);
 
-    c->queue = msgQ;
-    c->prev = this->clients;
-    this->clients = c;
+    client->queue = msgQ;
+    client->prev = this->clients;
+    this->clients = client;
 
     osSetIntMask(prevInt);
 
-    if (this->resetStatus > STATUS_IDLE) {
-        osSendMesg(c->queue, (OSMesg) & this->prenmiMsg, OS_MESG_NOBLOCK);
+    if (this->resetStatus > IRQ_RESET_IDLE) {
```

Comment:
```suggestion
    if (this->resetStatus >= IRQ_RESET_PRENMI) {
```
for symmetry with the next if